cli/cmd/status/status_test.go

```diff
@@ -28,39 +28,51 @@ import (
 	"k8s.io/client-go/kubernetes/fake"
 )
 
-// TestCheckConsulServers creates a fake stateful set and tests the checkConsulServers function.
-func TestCheckConsulServers(t *testing.T) {
-	c := getInitializedCommand(t, nil)
-	c.kubernetes = fake.NewSimpleClientset()
-
-	// First check that no stateful sets causes an error.
-	_, err := c.checkConsulServers("default")
-	require.Error(t, err)
-	require.Contains(t, err.Error(), "no server stateful set found")
-
-	// Next create a stateful set with 3 desired replicas and 3 ready replicas.
-	var replicas int32 = 3
+func TestCheckConsulAgents(t *testing.T) {
+	namespace := "default"
+	cases := map[string]struct {
+		desiredClients int
+		healthyClients int
+		desiredServers int
+		healthyServers int
+	}{
+		"No clients, no agents":                       {0, 0, 0, 0},
+		"3 clients - 1 healthy, no agents":            {3, 1, 0, 0},
+		"3 clients - 3 healthy, no agents":            {3, 3, 0, 0},
+		"3 clients - 1 healthy, 3 agents - 1 healthy": {3, 1, 3, 1},
+		"3 clients - 3 healthy, 3 agents - 3 healthy": {3, 3, 3, 3},
+		"No clients, 3 agents - 1 healthy":            {0, 0, 3, 1},
+		"No clients, 3 agents - 3 healthy":            {0, 0, 3, 3},
```

Comment:
I think we're using confusing terminology here, it should be `clients` and `servers` since `agents` means both :)